packages/engine/bin/cli/src/main.rs

```diff
@@ -73,8 +75,19 @@ pub struct SimpleExperimentArgs {
     experiment_name: ExperimentName,
 }
 
+#[derive(Debug)]
+pub struct CliError;
+
+impl fmt::Display for CliError {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt.write_str("Could not run engine")
```

Comment:
Isn't this just going to result in 

```
Could not run engine

Caused by
  Could not run engine
```

Reply:
No, it's the message which is printed when changing the context to `CliError`. Example output:
```
Error: Could not run engine
             at bin/cli/src/main.rs:133:10

Caused by:
   0: Engine didn't exit gracefully.
             at /home/ghost/Code/hash/packages/engine/lib/orchestrator/src/experiment.rs:441:9

Span trace:
   0: orchestrator::experiment::run
           with experiment_name=single_run experiment_id=bd4b4321-8e0a-4e00-b91f-4eafe99a89ae
             at lib/orchestrator/src/experiment.rs:217
```

Reply:
Yeah fair point we won't have a "Caused by" as they're two different processes. The reason I ask is you've got the same error message in the CLI and also in the `hash_engine` binary. And it feels like they should be different if we're actually trying to improve our error messages

Reply:
Hmm, do you have a suggestion that makes sense for the CLI? It's almost only using the orchestrator, nothing else.

Reply:
Could literally just go with "CLI encountered an error during execution"

```
Error: CLI encountered an error during execution
             at bin/cli/src/main.rs:133:10

Caused by:
   0: Engine didn't exit gracefully.
             at /home/ghost/Code/hash/packages/engine/lib/orchestrator/src/experiment.rs:441:9
```

I think we'd want to improve the message beyond just "CLI Error" if we start adding enumerations, but while it's a struct I don't think we have any other information to add